crates/gossip/src/initiate.rs

```diff
@@ -0,0 +1,646 @@
+use crate::gossip::K2Gossip;
+use crate::peer_meta_store::K2PeerMetaStore;
+use crate::K2GossipConfig;
+use kitsune2_api::agent::{AgentInfoSigned, LocalAgent};
+use kitsune2_api::peer_store::DynPeerStore;
+use kitsune2_api::{AgentId, DynLocalAgentStore, K2Result, Timestamp, Url};
+use rand::prelude::SliceRandom;
+use std::collections::HashSet;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::task::AbortHandle;
+
+pub fn spawn_initiate_task(
+    config: Arc<K2GossipConfig>,
+    gossip: K2Gossip,
+) -> AbortHandle {
+    tracing::info!("Starting initiate task");
+
+    let initiate_interval = config.initiate_interval();
+    let min_initiate_interval = config.min_initiate_interval();
+    tokio::task::spawn(async move {
+        loop {
+            tokio::time::sleep(initiate_interval).await;
+
+            match select_next_target(
```

Comment:
We're not checking if we have an active initiate round here. Is that because we assume it worst-case will have timed out during the initiate_interval period?